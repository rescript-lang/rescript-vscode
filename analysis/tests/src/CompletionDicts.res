// let dict = Js.Dict.fromArray([])
//                               ^com

// let dict = Js.Dict.fromArray([()])
//                                ^com

// let dict = Js.Dict.fromArray([("key", )])
//                                      ^com

// ^in+
let dict = Js.Dict.fromArray([
  ("key", true),
  //  ("key2", )
  //          ^com
])
// ^in-
