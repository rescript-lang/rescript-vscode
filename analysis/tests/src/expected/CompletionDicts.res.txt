Complete src/CompletionDicts.res 0:33
posCursor:[0:33] posNoWhite:[0:32] Found expr:[0:14->0:35]
Pexp_apply ...[0:14->0:31] (...[0:32->0:34])
Completable: Cexpression CArgument Value[Js, Dict, fromArray]($0)->array
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CArgument Value[Js, Dict, fromArray]($0)
ContextPath Value[Js, Dict, fromArray]
Path Js.Dict.fromArray
[{
    "label": "(_, _)",
    "kind": 12,
    "tags": [],
    "detail": "(key, 'a)",
    "documentation": null,
    "insertText": "(${1:_}, ${2:_})",
    "insertTextFormat": 2
  }]

Complete src/CompletionDicts.res 3:34
posCursor:[3:34] posNoWhite:[3:33] Found expr:[3:14->3:37]
Pexp_apply ...[3:14->3:31] (...[3:32->3:36])
Completable: Cexpression CArgument Value[Js, Dict, fromArray]($0)->array
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CArgument Value[Js, Dict, fromArray]($0)
ContextPath Value[Js, Dict, fromArray]
Path Js.Dict.fromArray
[{
    "label": "(_, _)",
    "kind": 12,
    "tags": [],
    "detail": "(key, 'a)",
    "documentation": null,
    "insertText": "(${1:_}, ${2:_})",
    "insertTextFormat": 2
  }]

Complete src/CompletionDicts.res 6:40
posCursor:[6:40] posNoWhite:[6:39] Found expr:[6:14->6:44]
Pexp_apply ...[6:14->6:31] (...[6:32->6:43])
Completable: Cexpression CArgument Value[Js, Dict, fromArray]($0)->array, tuple($1)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CArgument Value[Js, Dict, fromArray]($0)
ContextPath Value[Js, Dict, fromArray]
Path Js.Dict.fromArray
[]


Complete src/CompletionDicts.res 12:14
posCursor:[12:14] posNoWhite:[12:13] Found expr:[10:11->14:2]
Pexp_apply ...[10:11->10:28] (...[10:29->14:1])
Completable: Cexpression CArgument Value[Js, Dict, fromArray]($0)->array, tuple($1)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CArgument Value[Js, Dict, fromArray]($0)
ContextPath Value[Js, Dict, fromArray]
Path Js.Dict.fromArray
[{
    "label": "true",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }, {
    "label": "false",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }]


